### DISCUSSION TITLE
`x-forwarded-host` has port

### DISCUSSION THREAD

**Discussion by baloodevil** [vercel/next.js#68406] [Help]
### Summary

The JavaScript package I am using in my NextJs application expects `x-forwarded-host` to *not* contain the port (because it expects the port to be in `x-forwarded-port`).  

Unfortunately NextJs has the port in both places, which results in redirect errors to `localhost:3000:3000`.  While there is not a standard for whether or not `x-forwarded-host` contains the port, online examples and reason would say it should not, particularly if the port is included in the `x-forwarded-port` header.

![tempHeaders](https://github.com/user-attachments/assets/c92b440f-c8a4-4ddf-bd07-e01e2c0fd8f1)

**Comment by Netail** [comment]
According to the Mozilla specs;
`x-forwarded-host` has basically the same value specs as the `host` header, ~~which includes the port~~. Oops, optional indeed

**Reply by wingsofovnia** [reply]
I think the way it is written is confusing that leads to that conclusion.

If you go to MDN `Host` header [spec](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Host), port part is specifically denoted in `Syntax`, and it is not optional:
```
Host: <host>:<port>
```

What MDN says about `X-Forwarded-Host`:
> The HTTP X-Forwarded-Host (XFH) [request header](https://developer.mozilla.org/en-US/docs/Glossary/Request_header) is a de-facto standard header for identifying the original host requested by the client in the [Host](https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Host) HTTP request header.

I read as `X-Forwarded-Host` is a header for identifying the original `<host>` requested by the client in the Host header.

This is further reinforced by `Syntax` for `X-Forwarded-Host`:
```
X-Forwarded-Host: <host>
```

Which has no `<port>` part in it. Neither their example has a port.

At the same time, MDN has no `X-Forwarded-Port` which would complete that view.
